### DISCUSSION TITLE
Tasmota on ESP8266/ESP32: no chance to set ModBus baudrate over 76200 bps

### DISCUSSION THREAD

**Discussion by Radiomix2000** [arendst/Tasmota#16553] [Ideas]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I was need to set baudrate to 1150200 (console command: modbusbaudrate 115200), but it's not possible. When I try to set it, Console RESULT said it was set to 38400.
Experimentally I found, that 76200 - is maximum, what I can set and get adequate RESULT in console. If I set 77400 (increment 1200), it resets to 600 for some reason. It's the latest release (v12.1.1, e71235e), both tasmota.bin and tasmota32.bin. Also I compiled binary from source and had the same described problem.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _____ESP8266 and ESP32
- [x] Tasmota binary firmware version number used: _____v12.1.1
  - [X] Pre-compiled
  - [X] Self-compiled
- [X] Flashing tools used: _____esphome-flasher
- [X] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
11:15:01.824 RSL: RESULT = {"NAME":"ESP32-DevKit","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,0,1,1,1,0,0,0,0,1,1,1,1,1,0,0,1],"FLAG":0,"BASE":1}
11:15:02.045 RSL: RESULT = {"Module":{"1":"ESP32-DevKit"}}
11:15:02.309 RSL: RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"9408":"ModBr Tx"},"GPIO2":{"0":"None"},"GPIO3":{"9440":"ModBr Rx"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO24":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"0":"None"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [X] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
11:16:44.187 RSL: STATUS = {"Status":{"Module":1,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_58C740","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
11:16:44.200 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T01:31:42","StartupUTC":"2022-09-17T08:45:02","Sleep":50,"CfgHolder":4617,"BootCount":3,"BCResetTime":"2022-09-17T09:44:09","SaveCount":17}}
11:16:44.206 RSL: STATUS2 = {"StatusFWR":{"Version":"12.1.1(tasmota)","BuildDateTime":"2022-08-25T11:34:17","Core":"2_0_4_1","SDK":"v4.4.3","CpuFrequency":80,"Hardware":"ESP32-D0WDQ6 rev.1","CR":"402/699"}}
11:16:44.212 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Net",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
11:16:44.240 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":1414,"Free":1792,"Heap":146,"StackLowMark":3,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":3,"Features":["00000809","9F9AD7CF","0015A001","B7F7BFCF","05DA9BC0","E0360DC7","400840D2","20200000","D434482D","00000000"],"Drivers":"1,2,3,4,5,7,8,9,10,11,12,14,16,17,20,21,24,26,27,29,34,35,38,50,52,59,62,63,82,86,87","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,28,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,92,95,127"}}
11:16:44.250 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-58C740-1856","IPAddress":"192.168.0.131","Gateway":"192.168.0.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.0.1","DNSServer2":"0.0.0.0","Mac":"XX:XX:XX:XX:C7:40","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"0.0.0.0","DNSServer2":"0.0.0.0","Mac":"00:00:00:00:00:00"},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
11:16:44.257 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_58C740","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
11:16:44.264 RSL: STATUS7 = {"StatusTIM":{"UTC":"2022-09-17T10:16:44","Local":"2022-09-17T11:16:44","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"06:29","Sunset":"18:59"}}
11:16:44.279 RSL: STATUS10 = {"StatusSNS":{"Time":"2022-09-17T11:16:44","ESP32":{"Temperature":53.3},"TempUnit":"C"}}
11:16:44.291 RSL: STATUS11 = {"StatusSTS":{"Time":"2022-09-17T11:16:44","Uptime":"0T01:31:42","UptimeSec":5502,"Heap":145,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":40},"Wifi":{"AP":1,"SSId":"Net","BSSId":"14:CC:20:B4:CA:91","Channel":11,"Mode":"11n","RSSI":100,"Signal":-50,"LinkCount":1,"Downtime":"0T00:00:04"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
11:23:13.318 CMD: weblog
11:23:13.325 RSL: RESULT = {"WebLog":2}
11:24:15.366 CMD: weblog 4
11:24:15.373 RSL: RESULT = {"WebLog":4}
11:24:15.784 CFG: Saved, Count 19, Bytes 4096
11:24:20.312 CMD: modbusbaudrate 115200
11:24:20.315 SRC: WebConsole from 192.168.0.12
11:24:20.318 CMD: Grp 0, Cmd 'MODBUSBAUDRATE', Idx 1, Len 6, Pld 115200, Data '115200'
11:24:20.335 MBS: MBR SW ser init at 38400 baud
11:24:20.339 RSL: RESULT = {"ModbusBaudrate":38400}

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Write the command "modbusbaudrate 115200" in tasmota web-console

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
The baudrate is set to desired value and adequate RESULT in console

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
The log from console:
09:45:12.951 RSL: SENSOR = {"Time":"2022-09-17T09:45:12","ESP32":{"Temperature":53.3},"TempUnit":"C"}
09:45:26.364 CMD: modbusbaudrate
09:45:26.371 RSL: RESULT = {"ModbusBaudrate":9600}
09:45:29.239 CMD: modbusbaudrate 115200
09:45:29.260 RSL: RESULT = {"ModbusBaudrate":38400}
09:46:18.573 CMD: modbusbaudrate 76200
09:46:18.593 RSL: RESULT = {"ModbusBaudrate":76200}
09:46:21.679 CMD: modbusbaudrate 76300
09:46:21.687 RSL: RESULT = {"ModbusBaudrate":76200}
09:46:33.963 CMD: modbusbaudrate 115200
09:46:33.983 RSL: RESULT = {"ModbusBaudrate":38400}
09:48:25.543 CMD: modbusbaudrate 96000
09:48:25.562 RSL: RESULT = {"ModbusBaudrate":19200}
09:48:37.259 CMD: modbusbaudrate 77400
09:48:37.279 RSL: RESULT = {"ModbusBaudrate":600}
09:48:41.169 CMD: modbusbaudrate 76200
09:48:41.190 RSL: RESULT = {"ModbusBaudrate":76200}

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
See #16422
